### DISCUSSION TITLE
Tapo Cameras - unable to integrate

### DISCUSSION THREAD

**Discussion by ItsMeRitch** [koush/scrypted#1195] [Q&A]
**Describe the bug**
I had multiple Tapo camera models working perfectly with two-way-audio and ONVIF with motion detection. Since a recent update, two-way-audio stopped working. Decided to remove cameras and now unable to re-add to scrypted no matter what. 

When two-way-audio stopped working, logs from the homekit plugin were as follows:

[Tapo TC60] intercom failed to start RPCResultError [AxiosError]: Request failed with status code 401
  at settle (/node_modules/axios/dist/node/axios.cjs:1900:12)
  at v.<anonymous> (/node_modules/axios/dist/node/axios.cjs:2900:9)
  at v.emit (node:events:514:28)
  at v.emit (node:domain:489:12)
  at v._processResponse (/node_modules/follow-redirects/index.js:356:10)
  at ClientRequest._onNativeResponse (/node_modules/follow-redirects/index.js:62:10)
  at Object.onceWrapper (node:events:629:26)
  at ClientRequest.emit (node:events:514:28)
  at ClientRequest.emit (node:domain:489:12)
  at HTTPParser.parserOnIncomingClient (node:_http_client:700:27)
@scrypted/tapo:host
undefined error

After reloading all modules, updating scrypted and rolling back onvif and two-way-audio, nothing worked. 

Removed camera and now they are no longer automatically discovered and unable to add manually

![image](https://github.com/koush/scrypted/assets/55960162/e8da9395-2b90-48c4-b786-07c3efe4ee50)

Console logs from ONVIF plugin:

Error adding ONVIF camera Error: connect ECONNREFUSED 192.168.2.27:80
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1495:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '192.168.2.27',
  port: 80
}
Error adding ONVIF camera Error: connect ECONNREFUSED 192.168.2.15:80
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1495:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '192.168.2.15',
  port: 80
}

Devices are pingable from local networks and all functions accessible via tapo app 

No long/special characters passwords used

**Comment by ItsMeRitch** [comment]
Edit: Device RTSP Stream still accessible and working as normal with the exception of two-way-audio.  New error on the two-way-audio from homekit

[Tapo TC60] intercom failed to start RPCResultError: RTSP Error: RTSP/1.0 406 Not Acceptable
    at t.RtspClient.request (/common/src/rtsp-server.ts:605:19)
    at process.describe (node:internal/process/task_queues:95:5)
    at t.OnvifIntercom.intercomClient [as checkIntercom] (/src/onvif-intercom.ts:32:15)
    at t.OnvifIntercom.startIntercom (/src/onvif-intercom.ts:50:35)
    at RpcPeer.handleMessageInternal (/server/node_modules/@scrypted/server/src/rpc.ts:759:37)
@scrypted/onvif:host
host:@scrypted/homekit {
  cause: undefined

**Reply by ItsMeRitch** [reply]
Adding additional information: 

Tapo Two Way audio plugin installed and configured with cloud password, not local camera admin password. Plugin does not show any errors. 

![image](https://github.com/koush/scrypted/assets/55960162/0baf9ea4-84e7-4662-aaba-4fd4254a5d00)

![image](https://github.com/koush/scrypted/assets/55960162/b5fa9f3c-40c4-4ada-88a9-6576a48635b6)

**Comment by ion-storm** [comment]
same exact issue here

**Comment by koush** [comment]
need the tape plugin for two way audio

**Reply by ItsMeRitch** [reply]
You mean Tapo? It's already installed and configured.

**Comment by mishakim** [comment]
I'm having the same issue - brand new C110, created local account in the app, ECONNREFUSED in scripted. Updated the camera firmware to 1.3.9, no change.

console:
`Error adding ONVIF camera Error: connect ECONNREFUSED 10.0.1.42:80
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
  errno: -61,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '10.0.1.42',
  port: 80
}`

**Comment by mishakim** [comment]
Never mind - manually entered the port as 2020 (per [Tapo FAQ](https://www.tapo.com/us/faq/70/)) worked as it should.

**Comment by Littlesloths** [comment]
According to many ressources Tapo C200 cannot be used for ONVIF TALK, because Onvif protocol T is not implemented.
How did you get it working? I never got sound out of the camera...

**Reply by bmwcar** [reply]
C125 have same problem , have installed two way audio plugin ,the vedio is no problem , but it was no sound when i talk . But camera no problem when i talk in Tapo app

**Reply by sfnemis** [reply]
Tapo Two way audio is not working with homekit !

**Reply by opswhisperer** [reply]
Same, C120

**Reply by tellok** [reply]
> Same, C120

it's working on my c120, firmware version v1.1.15

**Reply by jarboer** [reply]
I got 2 way audio working on my C210 which is on the latest firmware. It didn't work until I enabled Third-Party Compatibility in the Tapo app under Me → Third-Party Services → Third-Party Compatibility. I also had to disable and re-enabled my Camera Account in the Tapo app → tap on the camera →tap on the device's settings icon → Advanced Settings → Camera Account. Once I did that it started working, it even works with internet access blocked on my firewall.

I also already had SSL setup for scrypted via a proxy (nginx proxy manager) using a self signed certificate, which some people in [this thread](https://github.com/AlexxIT/go2rtc/issues/1494) are saying is required.
